test/unit/org/apache/cassandra/net/ConnectionUtils.java

```diff
@@ -240,6 +247,31 @@ private void doCheck(FailCheck testAndFailCheck)
         }
     }
 
+    /**
+     * Perform the {@param assertion} within the specified duration repeatly until a success.
+     * Otherwise, it times out and re-throws the {@link AssertionError} from the {@param assertion}.
+     */
+    private static void longCheck(Runnable assertion, long timeout, TimeUnit timeUnit)
+    {
+        long startNano = System.nanoTime();
+        for (;;)
+        {
+            try
+            {
+                assertion.run();
+                return;
+            }
+            catch (AssertionError e)
+            {
+                long elapsedNano = System.nanoTime() - startNano;
+                if (elapsedNano > timeUnit.toNanos(timeout))
```

Comment:
sorry I didn't comment sooner, I left it in my example; it would be best to compute `timeUnit.toNanos(timeout)` once outside the loop